packages/commonwealth/server/passport/magic.ts

```diff
@@ -396,6 +400,21 @@ async function magicLoginRoute(
   // canonical address is always ethereum
   const canonicalAddress = decodedMagicToken.publicAddress;
 
+  // replay attack check
+  const didTokenId = decodedMagicToken.claim.tid; // single-use token id
+  const previouslyUsedSsoTokenByTID = await models.SsoToken.findOne({
+    where: {
+      state_id: didTokenId,
+    },
+  });
+
+  if (previouslyUsedSsoTokenByTID) {
+    log.warn('Replay attack detected.');
+    throw new Error(
+      `Replay attack detected for user ${decodedMagicToken.publicAddress}}.`,
+    );
+  }
```

Comment:
If attacker attempts to replay Magic token, it will be rejected because we keep track of magic single-use ID in SSO Token.

Reply:
IMO we should destroy the SSOToken object once used, vs maintain it forever.

Reply:
Now using `Address.verification_token` instead. Once a Magic token is verified, and it is not already being used, it will replace the previous `verification_token`.

That way we can leave SSOTokens alone here, or choose to destroy them.